packages/react-components/react-combobox/src/utils/useComboboxPopup.ts

```diff
@@ -31,8 +34,7 @@ export function useComboboxPopup(
 
   const { targetRef, containerRef } = usePositioning(popperOptions);
 
-  return [
-    { ...triggerShorthand, ref: useMergedRefs(triggerShorthand?.ref, targetRef) },
-    { ...listboxShorthand, ref: useMergedRefs(listboxShorthand?.ref, containerRef) },
-  ];
+  const listbox = listboxShorthand && { ...listboxShorthand, ref: useMergedRefs(listboxShorthand?.ref, containerRef) };
```

Comment:
super-nit: the `?.` isn't needed because you already checked for undefined at the start of the line
```suggestion
  const listbox = listboxShorthand && { ...listboxShorthand, ref: useMergedRefs(listboxShorthand.ref, containerRef) };
```

Reply:
needed to move it out of the conditional anyway, since it needs to be consistent across renders :P